src/libs/Navigation/AppNavigator/ReportScreenIDSetter.ts

```diff
@@ -0,0 +1,80 @@
+import {useEffect} from 'react';
+import {OnyxCollection, OnyxEntry, withOnyx} from 'react-native-onyx';
+import usePermissions from '@hooks/usePermissions';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as App from '@userActions/App';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Policy, Report} from '@src/types/onyx';
+import type {ReportScreenWrapperProps} from './ReportScreenWrapper';
+
+type ReportScreenIDSetterComponentProps = {
+    /** Available reports that would be displayed in this navigator */
+    reports: OnyxCollection<Report>;
+
+    /** The policies which the user has access to */
+    policies: OnyxCollection<Policy>;
+
+    /** Whether user is a new user */
+    isFirstTimeNewExpensifyUser: OnyxEntry<boolean>;
```

Comment:
@blazejkustra I've checked different usages of `initialValue` field in the E/App code and found it can be
`initialValue: null`, so updating initialValue type in the onyx lib maybe not the best option.
But we can use selector instead of `initialValue` for example, [ReportScreenIDSetter.ts](https://github.com/Expensify/App/pull/29157/files/274073d76e5782bcf49ce4b17cc892697e051faf#diff-0cc4d9a3611c71f41758ac8f941812b2ca17e11c8c37f9689fb320f32eff2644) file:
```
isFirstTimeNewExpensifyUser: {
        key: ONYXKEYS.NVP_IS_FIRST_TIME_NEW_EXPENSIFY_USER,
        selector: value => value ?? false,
    },
```
So this way, we can have just a boolean in type without an error
```
 isFirstTimeNewExpensifyUser: boolean;
```

However, we need to check that it doesn't affect the performance because I've looked through a couple of commits and initialValues were added for performance optimization.

Reply:
Yees, let's not touch it in TS migration 👍

Reply:
Yes, it sets the initial value but I think it doesn't prevent you from setting it to `null` later, so it could still be null 🤔